### PR TITLE
[ML] fixing ML composite agg restart test

### LINKED ISSUE #76075: [CI] DatafeedJobsIT testStopAndRestartCompositeDatafeed failing
The assertion is that the event counts for the scroll buckets and composite buckets are the same, but the more fundamental problem that the extra debug information shows up is that the _timestamp_ of the mismatched bucket is different - we're not comparing like with like for some reason.

**Build scan:**
https://gradle-enterprise.elastic.co/s/63sdskgd2dbki/tests/:x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest/org.elasticsearch.xpack.ml.integration.DatafeedJobsIT/testStopAndRestartCompositeDatafeed

**Reproduction line:**
`./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.ml.integration.DatafeedJobsIT.testStopAndRestartCompositeDatafeed" -Dtests.seed=BCBE2D3EFDF6C14B -Dtests.locale=en-ZA -Dtests.timezone=Pacific/Samoa -Druntime.java=11 -Dtests.fips.enabled=true`

**Applicable branches:**
7.x, 7.15, master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.ml.integration.DatafeedJobsIT&tests.test=testStopAndRestartCompositeDatafeed

**Failure excerpt:**
```
java.lang.AssertionError: Hits: {
  "_index" : "stop-restart-data",
  "_type" : "type",
  "_id" : "BSwfDXsBsYrSD3fRB57C",
  "_score" : 1.0,
  "_source" : {
    "time" : 1626804286772,
    "@timestamp" : 1626804286772
  }
}
{
  "_index" : "stop-restart-data",
  "_type" : "type",
  "_id" : "XiwfDXsBsYrSD3fRB57C",
  "_score" : 1.0,
  "_source" : {
    "time" : 1626804000445,
    "@timestamp" : 1626804000445
  }
}
{
  "_index" : "stop-restart-data",
  "_type" : "type",
  "_id" : "5SwfDXsBsYrSD3fRB57C",
  "_score" : 1.0,
  "_source" : {
    "time" : 1626805874948,
    "@timestamp" : 1626805874948
  }
}
{
  "_index" : "stop-restart-data",
  "_type" : "type",
  "_id" : "VCwfDXsBsYrSD3fRB5_C",
  "_score" : 1.0,
  "_source" : {
    "time" : 1626806183402,
    "@timestamp" : 1626806183402
  }
}
{
  "_index" : "stop-restart-data",
  "_type" : "type",
  "_id" : "ViwfDXsBsYrSD3fRB5_C",
  "_score" : 1.0,
  "_source" : {
    "time" : 1626805948657,
    "@timestamp" : 1626805948657
  }
} 
 failure: composite bucket [Tue Jul 20 08:00:00 SST 2021] [2] does not equal scroll bucket [Tue Jul 20 07:00:00 SST 2021] [5]
Expected: <5L>
     but: was <2L>

  at __randomizedtesting.SeedInfo.seed([BCBE2D3EFDF6C14B:D200B6AEC903AC84]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.elasticsearch.xpack.ml.integration.DatafeedJobsIT.testStopAndRestartCompositeDatafeed(DatafeedJobsIT.java:470)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:834)

```

### PR DESCRIPTION
**Pull request by benwtrent**
The test failure is due to data generation. This isn't caused by a bug, but is instead a test set up failure.

The issue is that scroll is seeing the data as a partial result, and aggs is not. Consequently, the first bucket is sometimes only present in scroll, but missing in aggs.

This is acceptable.

This commit fixes the test to make it more deterministic.

closes https://github.com/elastic/elasticsearch/issues/76075

### CODE REVIEW

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java [hunk 1]
```diff
@@ -309,7 +311,7 @@ public void testStopAndRestartCompositeDatafeed() throws Exception {
             .setMapping("time", "type=date")
             .get();
         long numDocs = randomIntBetween(32, 2048);
-        long now = System.currentTimeMillis();
+        long now = Intervals.alignToFloor(System.currentTimeMillis(), TimeValue.timeValueHours(1).millis()) + 1;
```
Comment (benwtrent) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java:314]: this is still causing a failure :(((. Trying to figure out how to make this deterministic.

Comment (benwtrent) [reply x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java]: Now fixed. Making sure all timestamps are not on the edge of buckets given the start/finish/ and interval.

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/support/BaseMlIntegTestCase.java [hunk 2]
```diff
@@ -315,6 +316,38 @@ public static void indexDocs(Logger logger, String index, long numDocs, long sta
         logger.info("Indexed [{}] documents", numDocs);
     }
 
+    public static void indexDocs(Logger logger, String index, long numDocs, long start, long end, TimeValue interval) {
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/support/BaseMlIntegTestCase.java:319]: I am a bit worried about adding a second version of `indexDocs` here. I think the problem with the test is that we need to ensure the first bucket is not partial. Could we not change the existing `indexDocs` to always index the first record with `start` time and then change the test to do the bucket alignment?

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Pull request merged**